src/structui/ui.py

```
@@ -275,14 +275,23 @@ def delete_current_container():
             props_container = ui.column().classes('w-full gap-4')
             
             def render_primitive_input(k, v, parent_node):
-                def make_on_change(prop_key=k):
+                meta = self.schema_manager.get_meta(str(k))
+
+                def make_on_change(prop_key=k, prop_type=meta.get('type')):
```

Ruff (0.15.12)

[warning] Do not perform function call `meta.get` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)